python/ptest/cases/gomspace_checkout_case.py

```diff
@@ -133,29 +133,36 @@ def write_state(self, string_state, state_value):
         if ppt_mode_cmd == ppt_mode_updated:
             print("Could not update pptmode")
 
-        heater_cmd = self.str_to_bool(read_state(self, "gomspace.heater_cmd"))
-        heater_cmd_updated = self.str_to_bool(write_state(
-            self, "gomspace.heater_cmd", not heater_cmd))
-        if heater_cmd == heater_cmd_updated:
-            print("Could not update heater")
+        heater_init = self.str_to_bool(read_state(self, "gomspace.heater"))
+        write_state(self, "gomspace.heater_cmd", str(not heater_init).lower())
+        heater_updated=self.str_to_bool(read_state(self, "gomspace.heater"))
+
+        cycle_init = int(read_state(self, "pan.cycle_no"))
+        cycle_no = int(read_state(self, "pan.cycle_no"))
+        while(cycle_no-cycle_init<100 and heater_init==heater_updated):
+            cycle_no = int(read_state(self, "pan.cycle_no"))
+            heater_updated=self.str_to_bool(read_state(self, "gomspace.heater"))
+
+        if (heater_init==heater_updated):
+            print("could not update heater")
 
         counter_reset_cmd = self.str_to_bool(read_state(
             self, "gomspace.counter_reset_cmd"))
         counter_reset_cmd_updated = self.str_to_bool(write_state(
-            self, "gomspace.counter_reset_cmd", not counter_reset_cmd))
```

Comment:
Shouldn't you check the counter values to see if the counter(s) resetted?

Reply:
Yeah I know, for some reason the counter values aren't changing on ptest when I call reset. I didn't get to finish looking into why that is the case with the reset command today. I just wanted to fix the write state call in this pr, I was planning on fixing the rest of this test during the next work session?

Reply:
If the test isn't ready to be finalized, let's open the PR later when it is.